// tiff/lib/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.tagNames = exports.TiffIfd = void 0;
exports.decode = decodeTIFF;
exports.isMultiPage = isMultiPage;
exports.pageCount = pageCount;
const exif_1 = require("./tags/exif");
const gps_1 = require("./tags/gps");
const standard_1 = require("./tags/standard");
const tiffDecoder_1 = __importDefault(require("./tiffDecoder"));
const tiffIfd_1 = __importDefault(require("./tiffIfd"));
exports.TiffIfd = tiffIfd_1.default;
function decodeTIFF(data, options) {
    const decoder = new tiffDecoder_1.default(data);
    return decoder.decode(options);
}
function isMultiPage(data) {
    const decoder = new tiffDecoder_1.default(data);
    return decoder.isMultiPage;
}
function pageCount(data) {
    const decoder = new tiffDecoder_1.default(data);
    return decoder.pageCount;
}
const tagNames = {
    exif: exif_1.tagsById,
    gps: gps_1.tagsById,
    standard: standard_1.tagsById,
};
exports.tagNames = tagNames;
